### DISCUSSION TITLE
Steps to transfer a pre-trained model weights to new model for warmed up start.

### DISCUSSION THREAD

**Discussion by p0p4k** [coqui-ai/TTS#2415] [General]
Load the models using the TTS models (eg. Vits(...)). This script is helpful to load the already pretrained wieghts in the model zoo and use them. 
Now if you have a pretrained model, without speaker encoder, but you want to transfer the weights to a new model which uses speaker encoder, a few layers will be different here and there. Some might have name mismatches and some might have shape mismatches. The code below will help anyone to load the pretrained weights into appropriate layers and help kick start the training. Any questions, please ask.
```
pretrained_dict = pre_trained_model.state_dict()
model_dict = model.state_dict() 

processed_dict = {}

for k in model_dict.keys(): 
    if k in pretrained_dict.keys():
        if pretrained_dict[k].shape == model_dict[k].shape:
            processed_dict[k] = pretrained_dict[k] 
        else:
            processed_dict[k] = model_dict[k]
    else:
        processed_dict[k] = model_dict[k]

model.load_state_dict(processed_dict, strict=False)
```

**Comment by p0p4k** [comment]
This can be made a part of the regular training script with little effort, by matching the tts model name and  dataset language, we can automagically load a pretrained model from the zoo for warmup start. (this is slightly different from restore model)
